dask_sql/mappings.py

```diff
@@ -68,7 +75,8 @@
 _SQL_TO_PYTHON_FRAMES = {
     "SqlTypeName.DOUBLE": np.float64,
     "SqlTypeName.FLOAT": np.float32,
-    "SqlTypeName.DECIMAL": np.float64,  # We use np.float64 always, even though we might be able to use a smaller type
+    # a column of Decimals in panda is `object`, but cuDF has a dedicated dtype
+    "SqlTypeName.DECIMAL": object if not cudf else cudf.Decimal128Dtype(38, 10),
```

Comment:
What happens in cases where `cudf` is importable but users want to still run on the CPU?

Reply:
Handling still needs to be added there, as in that case we would end up failing on CPU; think the best choice for now is adding a Dask config variable but once #1117 is further along it would probably make sense to base this on whether or not we're working with a GPU/CPU context

Reply:
In the interest of this feature making it in, I'm okay with merging this in as is and following up with a config option or #1117 before the next release.